include/eve/module/algo/algo/for_each_iteration_with_expensive_optional_part.hpp

```diff
@@ -0,0 +1,259 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/module/algo/algo/concepts.hpp>
+#include <eve/module/algo/algo/traits.hpp>
+#include <eve/module/core.hpp>
+
+namespace eve::algo
+{
+
+namespace detail
+{
+  struct for_each_iteration_with_expensive_optional_part_common
+  {
+    template<typename Traits, typename I, typename S> auto unroll_l(Traits, I f, S l)
+    {
+      return eve::unalign(f) + (l - f - get_unrolling<Traits>() * iterator_cardinal_v<I>);
+    }
+
+    template<typename Traits, typename I, typename S, typename Delegate>
+    EVE_FORCEINLINE bool no_unrolling_loop(Traits, I& f, S l, Delegate& delegate) const
+    {
+      while( f < l )
+      {
+        if( delegate.step(f, eve::ignore_none) ) return true;
+        f += iterator_cardinal_v<I>;
+      }
+      return false;
+    }
+
+    template<typename Traits, typename I, typename S, typename Delegate>
+    EVE_FORCEINLINE bool main_loop(Traits tr, I& f, auto unroll_l, S l, Delegate& delegate) const
+    requires(get_unrolling<Traits>() == 1)
+    {
+      (void)unroll_l;
```

Comment:
You can mark the paramter [[maybe_unused]], it is better than random cast to void IMO.

```
EVE_FORCEINLINE bool main_loop(Traits tr, I& f, [[maybe_unused]] auto unroll_l, S l, Delegate& delegate) const
```

Reply:
Any update ont hsi samll details ?
The PR is fine once fixed and can be merged

Reply:
1. I don't have time to fix this until the weekend.
2. I still want to add more testing for iteration.

It's good if everything else seems ok.